### PR TITLE
add(http): add http cache and catalog

### PR DESCRIPTION
**Pull request by USTC-Han**
- 添加HTTP缓存
- 填加HTTP目录

### CODE REVIEW

File: docs/http/5.http-cache/browser-caches.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+为了更好的理解这几个字段的意思，可以看下面这张流程图：
+
+<Img width="525" legend="图：服务器的缓存控制" src="https://cosmos-x.oss-cn-hangzhou.aliyuncs.com/oDMgW4.png"/>
+
+#### 客户端的缓存控制：
+
+`Cache-Control`在客户端的缓存控制中常用的参数为`max-age`和`no-cache`两个。
+
+当我们点击刷新按钮、使用`CMD + R`、`F5`“刷新”网页的时候，有的请求头中会添加`Cache-Control：max-age=0`字段。因为 `max-age` 是“生存时间”，而本地缓存里的数据至少保存了几秒钟，所以`max-age=0`就意味着**不直接读取浏览器缓存，必须向服务器发出请求**。服务器看到`Cache-Control: max-age=0`，此时服务器就会**判断协商缓存**的相关字段，如果命中协商缓存，返回 304 状态码，读取本地缓存。如果没有命中协商缓存，就返回新的资源给客户端。如下图：
```
Comment (Kimi-Gao) [inline docs/http/5.http-cache/browser-caches.md]: Adding a snapshot as following:
image

File: docs/http/5.http-cache/browser-caches.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+<Img width="800" legend="图：非强制刷新" src="https://cosmos-x.oss-cn-hangzhou.aliyuncs.com/2coR6N.png"/>
+
+当我们使用`CMD + Shift + R`或`Cache + F5`“强制刷新”网页的时候，我们可以从请求头中看到`Cache-Control：no-cache`的字段。此时意味着不使用任何缓存数据，服务器必须返回新的资源给客户端。如下图：
```
Comment (Kimi-Gao) [inline docs/http/5.http-cache/browser-caches.md]: What does `Cache + F5` mean?
And what's the diff between `disable cache + CMD R` and `CMD + Shift + R`

Comment (Kimi-Gao) [reply docs/http/5.http-cache/browser-caches.md]: image

Comment (USTC-Han) [reply docs/http/5.http-cache/browser-caches.md]: 这块我不太确定，我觉得两者没有什么区别

**Pull request merged**